File: server/hotrod/src/test/scala/org/infinispan/server/hotrod/test/TestClientListener.scala
package org.infinispan.server.hotrod.test

import org.infinispan.notifications.cachelistener.event.Event
import org.infinispan.server.hotrod.Bytes

/**
 * @author Galder Zamarreño
 */
trait TestClientListener {

   def onCreated(event: TestKeyWithVersionEvent) {} // no-op
   def onModified(event: TestKeyWithVersionEvent)  {} // no-op
   def onRemoved(event: TestKeyEvent)  {} // no-op
   def onCustom(event: TestCustomEvent)  {} // no-op

   def queueSize(eventType: Event.Type): Int = 0
   def pollEvent(eventType: Event.Type): AnyRef = null

   def customQueueSize(): Int = 0
   def pollCustom(): TestCustomEvent = null

   def getId : Bytes

}

File: server/hotrod/src/test/java/org/infinispan/server/hotrod/test/TestClientListener.java
package org.infinispan.server.hotrod.test;

import org.infinispan.notifications.cachelistener.event.Event;

/**
 * @author Galder Zamarreño
 */
public abstract class TestClientListener {

   public void onCreated(TestKeyWithVersionEvent event) {
   } // no-op

   public void onModified(TestKeyWithVersionEvent event) {
   } // no-op

   public void onRemoved(TestKeyEvent event) {
   } // no-op

   public void onCustom(TestCustomEvent event) {
   } // no-op

   public int queueSize(Event.Type eventType) {
      return 0;
   }

   public Object pollEvent(Event.Type eventType) {
      return null;
   }

   public int customQueueSize() {
      return 0;
   }

   public TestCustomEvent pollCustom() {
      return null;
   }

   public abstract byte[] getId();
}
